AppController/djinn.rb

```diff
@@ -5865,6 +5876,81 @@ def get_node_stats_json(secret)
     JSON.dump(node_stats)
   end
 
+  # Gets summarized total_requests, total_req_in_queue and current_sessions 
+  # for a specific application version accross all LB nodes.
+  #
+  # Args:
+  #   version_key: A string specifying the version key.
+  #
+  def get_application_load_stats(version_key)
+    total_requests, requests_in_queue, sessions = 0, 0, 0
+    pxname = "gae_#{version_key}"
+    time = :no_stats
+    lb_nodes = @nodes.select{|node| node.is_load_balancer?}
+    lb_nodes.each{ |node|
+      begin
+        ip = node.private_ip
+        load_stats = HermesClient.get_proxy_load_stats(ip, @@secret, pxname)
+        total_requests += load_stats[0]
+        requests_in_queue += load_stats[1]
+        sessions += load_stats[2]
+        time = Time.now.to_i
+      rescue AppScaleException => error
+        Djinn.log_warn("Couldn't get proxy stats from Hermes: #{error.message}")
+      end
+    }
+    if lb_nodes.length > 1
+      # Report total HAProxy stats if there are multiple LB nodes
+      Djinn.log_debug("Summarized HAProxy load stats for #{pxname}: " \
+        "req_tot=#{total_requests}, qcur=#{requests_in_queue}, scur=#{sessions}")
+    end
+    return total_requests, requests_in_queue, sessions, time
+  end
+
+  # Gets united lists of running and failed AppServers
+  # for a specific application version accross all LB nodes.
+  #
+  # Args:
+  #   version_key: A string specifying the version key.
+  #
+  def get_application_appservers(version_key)
+    all_running, all_failed = [], []
+    pxname = "gae_#{version_key}"
+    lb_nodes = @nodes.select{|node| node.is_load_balancer?}
+    lb_nodes.each{ |node|
+      begin
+        ip = node.private_ip
+        running, failed = HermesClient.get_backend_servers(ip, @@secret, pxname)
+        all_running += running
+        all_failed += failed
+      rescue AppScaleException => error
+        Djinn.log_warn("Couldn't get proxy stats from Hermes: #{error.message}")
+      end
+    }
+    all_running.uniq!
+    all_failed.uniq!
+
+    if lb_nodes.length > 1
```

Comment:
@whoarethebritons `get_proxy_load_stats` function which is called for every LB node writes own logs.. this summary is needed only in case of multiple LBs.